src/main/services/web-torrent-data.ts

```diff
@@ -0,0 +1,32 @@
+import WebTorrentHealth from "webtorrent-health";
+
+type WebTorrentHealthData = {
+  seeds: number;
+  peers: number;
+};
+
+export const webTorrentData = {
+  async getSeedersAndPeers(magnet: string, retry = 0, timeout = 1500) {
+    return new Promise((resolve, reject) => {
+      WebTorrentHealth(
+        magnet,
+        { timeout },
+        (err: Error, data: WebTorrentHealthData) => {
+          if (err) {
+            return reject(err);
+          }
+
+          const { peers, seeds } = data;
+
+          if ((!peers || !seeds) && retry < 3) {
+            return resolve(
+              webTorrentData.getSeedersAndPeers(magnet, retry + 1, timeout * 2)
+            );
+          }
+
+          return resolve({ peers, seeders: seeds });
+        }
+      );
+    });
+  },
```

Comment:
i don't like the recursive function

could you replace with a loop solution?
maybe something like this (just an example):

```
  async getSeedersAndPeers(magnet: string) {
    let peers = 0;
    let seeds = 0;
    let retry = 0;

    while (retry < 3 && (!peers || !seeds)) {
      try {
        const data = await webTorrentHealth(magnet, {
          timeout: 1500 * (retry + 1),
        });

        peers = data.peers;
        seeds = data.seeds;
      } catch (e) {
        retry++;
      }
    }

    return { peers, seeders: seeds };
  },
```

Reply:
Done